### PR TITLE
GALAXY-1745_Article_Config_Integration 

### PR DESCRIPTION
**Pull request by TomMarshBBC**
…ile. Added support for new ArticleUseCaseConfig.

### CODE REVIEW

File: app/src/main/java/bbc/co/uk/nautilusapp/di/modules/interactors/LayoutInteractorModule.kt [hunk 1]
```diff
@@ -43,17 +40,11 @@ class LayoutInteractorModule {
...
     ): Repository<String, FetchOptions, LayoutResponse> {
         return RepositoryBuilder(OkHttpNetworkSource(client, LogIfOnMainThreadCheck()), deserialiser)
             .cache(CacheWithTTL({ System.currentTimeMillis() }))
-            .optionModifier(
-                EndpointFlagpoleModifier(
-                    remoteConfigInteractor.fetchEndpointStatus(EndPointType.LAYOUT),
-                    EndPointType.LAYOUT.id
-                )
-            )
+            //.optionModifier(/**EndpointFlagpoleModifier(remoteConfigInteractor.fetchEndpointStatus(EndPointType.LAYOUT),EndPointType.LAYOUT.id)**/)
```
Comment (johnryM) [inline app/src/main/java/bbc/co/uk/nautilusapp/di/modules/interactors/LayoutInteractorModule.kt]: you still need this line?

File: app/src/main/java/bbc/co/uk/nautilusapp/di/modules/interactors/TrevorArticleInteractorModule.kt [hunk 2]
```diff
@@ -25,40 +25,43 @@ import uk.co.bbc.colca.source.okhttp.OkHttpNetworkSource
...
     @Named("network-repository")
-    fun provideNetworkRepository(client: OkHttpClient, @Named("response-extractor") deserialiser: Repository.Deserialiser<TrevorArticleResponse>,
-                                 remoteConfigInteractor: RemoteConfigInteractor): Repository<String, FetchOptions, TrevorArticleResponse> {
+    fun provideNetworkRepository(client: OkHttpClient, @Named("response-extractor") deserialiser: Repository.Deserialiser<TrevorArticleResponse>): Repository<String, FetchOptions, TrevorArticleResponse> {
         return RepositoryBuilder(OkHttpNetworkSource(client, LogIfOnMainThreadCheck()), deserialiser)
-                .cache(CacheWithTTL({ System.currentTimeMillis() }))
-                .optionModifier(EndpointFlagpoleModifier(remoteConfigInteractor.fetchEndpointStatus(EndPointType.CONTENT), EndPointType.CONTENT.id))
-                .build()
+            .cache(CacheWithTTL({ System.currentTimeMillis() }))
+            //.optionModifier(/**EndpointFlagpoleModifier(remoteConfigInteractor.fetchEndpointStatus(EndPointType.CONTENT), EndPointType.CONTENT.id)**/)
```
Comment (johnryM) [inline app/src/main/java/bbc/co/uk/nautilusapp/di/modules/interactors/TrevorArticleInteractorModule.kt]: this too?

File: app/src/main/java/bbc/co/uk/nautilusapp/di/modules/interactors/TrevorIndexInteractorModule.kt [hunk 3]
```diff
@@ -45,17 +41,11 @@ object TrevorIndexInteractorModule {
...
     ): Repository<String, FetchOptions, TrevorIndexResponse> {
         return RepositoryBuilder(OkHttpNetworkSource(client, LogIfOnMainThreadCheck()), deserialiser)
             .cache(CacheWithTTL({ System.currentTimeMillis() }))
-            .optionModifier(
-                EndpointFlagpoleModifier(
-                    remoteConfigInteractor.fetchEndpointStatus(EndPointType.CONTENT),
-                    EndPointType.CONTENT.id
-                )
-            )
+            //.optionModifier(/**EndpointFlagpoleModifier(remoteConfigInteractor.fetchEndpointStatus(EndPointType.CONTENT),EndPointType.CONTENT.id)**/)
```
Comment (johnryM) [inline app/src/main/java/bbc/co/uk/nautilusapp/di/modules/interactors/TrevorIndexInteractorModule.kt]: this too?

File: app/src/main/java/bbc/co/uk/nautilusapp/di/modules/interactors/LayoutInteractorModule.kt (hunk 1 above)
Comment (TomMarshBBC) [reply app/src/main/java/bbc/co/uk/nautilusapp/di/modules/interactors/LayoutInteractorModule.kt]: I don't know to be honest with you, depends how we end up doing all the remote config stuff in Nautilus

**Pull request merged**